src/Vipps.net.AspCoreDemo/Startup.cs

```diff
@@ -21,7 +22,29 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            var vippsConfigurationOptions = new VippsConfigurationOptions
+            {
+                ClientId = Configuration.GetValue<string>("CLIENT-ID")!,
+                ClientSecret = Configuration.GetValue<string>("CLIENT-SECRET")!,
+                MerchantSerialNumber = Configuration.GetValue<string>("MERCHANT-SERIAL-NUMBER")!,
+                SubscriptionKey = Configuration.GetValue<string>("SUBSCRIPTION-KEY")!,
+                UseTestMode = true,
+                PluginName = Assembly.GetExecutingAssembly().GetName().Name,
+                PluginVersion =
+                    Assembly.GetExecutingAssembly().GetName().Version?.ToString() ?? "1.0.0"
+            };
+
+            services.AddTransient(_ => vippsConfigurationOptions);
+            services.AddTransient<IVippsApi, VippsApi>(
+                (sp) =>
+                    new VippsApi(vippsConfigurationOptions, null, sp.GetService<ILoggerFactory>())
+            );
+
             services.AddControllers();
+            services.AddSwaggerGen(c =>
+            {
+                c.SwaggerDoc("v1", new OpenApiInfo { Title = "Your API Name", Version = "v1" });
+            });
```

Comment:
Does swagger have to be here? if so, maybe change the title from "your api name"

Reply:
Added by the summer students. It does make it easy to test, so I've left it. Yeah can change the title :)